lib/Elastica/Search.php

```diff
@@ -163,73 +144,53 @@ public function setOptions(array $options)
         return $this;
     }
 
-    /**
-     * @return $this
-     */
-    public function clearOptions()
+    public function clearOptions(): self
     {
-        $this->_options = [];
+        $this->options = [];
 
         return $this;
     }
 
     /**
-     * @param string $key
-     * @param mixed  $value
-     *
-     * @return $this
+     * @param mixed $value
      */
-    public function addOption($key, $value)
+    public function addOption(string $key, $value): self
     {
-        $this->_validateOption($key);
+        $this->validateOption($key);
 
-        $this->_options[$key][] = $value;
+        $this->options[$key][] = $value;
 
         return $this;
     }
 
-    /**
-     * @param string $key
-     *
-     * @return bool
-     */
-    public function hasOption($key)
+    public function hasOption(string $key): bool
     {
-        return isset($this->_options[$key]);
+        return isset($this->options[$key]);
     }
 
     /**
-     * @param string $key
-     *
-     * @throws InvalidException
+     * @throws InvalidException if the given key does not exists as an option
      *
      * @return mixed
      */
-    public function getOption($key)
+    public function getOption(string $key)
     {
         if (!$this->hasOption($key)) {
             throw new InvalidException('Option '.$key.' does not exist');
         }
 
-        return $this->_options[$key];
+        return $this->options[$key];
     }
 
-    /**
-     * @return array
-     */
-    public function getOptions()
+    public function getOptions(): array
     {
-        return $this->_options;
+        return $this->options;
     }
 
     /**
-     * @param string $key
-     *
-     * @throws InvalidException
-     *
-     * @return bool
+     * @throws InvalidException If the given key is not a valid option
      */
-    protected function _validateOption($key)
+    private function validateOption(string $key): void
```

Comment:
change to private on purpose? I think it's fine just want to make sure it is on purpose?

Reply:
Moved back to "protected". I don't see why we would allow to extend from this class, tho